pkg/generate/app/app.go

```diff
@@ -109,32 +111,46 @@ func (r *SourceRef) SuggestName() (string, bool) {
 	if len(r.Name) > 0 {
 		return r.Name, true
 	}
-	return nameFromGitURL(r.URL)
+	if r.URL != nil {
+		return nameFromGitURL(r.URL)
+	}
+	return "", false
 }
 
 // BuildSource returns an OpenShift BuildSource from the SourceRef
 func (r *SourceRef) BuildSource() (*buildapi.BuildSource, []buildapi.BuildTriggerPolicy) {
-	return &buildapi.BuildSource{
+	triggers := []buildapi.BuildTriggerPolicy{
+		{
+			Type: buildapi.GitHubWebHookBuildTriggerType,
+			GitHubWebHook: &buildapi.WebHookTrigger{
+				Secret: generateSecret(20),
+			},
+		},
+		{
+			Type: buildapi.GenericWebHookBuildTriggerType,
+			GenericWebHook: &buildapi.WebHookTrigger{
+				Secret: generateSecret(20),
+			},
+		},
```

Comment:
Good point.  Maybe you don't want it?  But we should probably use the
defaulter on a lot of other stuff.  We could probably reuse the secret too.


On Sep 17, 2015, at 8:17 PM, Ben Parees <notifications@github.com> wrote:

In pkg/generate/app/app.go
<https://github.com/openshift/origin/pull/4717#discussion_r39814161>:

>  }
>
>  // BuildSource returns an OpenShift BuildSource from the SourceRef
>  func (r *SourceRef) BuildSource() (*buildapi.BuildSource, []buildapi.BuildTriggerPolicy) {
> -	return &buildapi.BuildSource{
> +	triggers := []buildapi.BuildTriggerPolicy{
> +		{
> +			Type: buildapi.GitHubWebHookBuildTriggerType,
> +			GitHubWebHook: &buildapi.WebHookTrigger{
> +				Secret: generateSecret(20),
> +			},
> +		},
> +		{
> +			Type: buildapi.GenericWebHookBuildTriggerType,
> +			GenericWebHook: &buildapi.WebHookTrigger{
> +				Secret: generateSecret(20),
> +			},
> +		},

why don't we add genericwebhooktrigger to all builds, just like we add
configchangetrigger?

i can understand not adding github, but generic could be useful for any
build type. (i know, not introduced by your change, but since you're moving
it around......)

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4717/files#r39814161>.